satellite/metainfo/validation.go

```diff
@@ -0,0 +1,257 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package metainfo
+
+import (
+	"bytes"
+	"context"
+	"sync"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"storj.io/storj/pkg/auth"
+	"storj.io/storj/pkg/macaroon"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/console"
+)
+
+const requestTTL = time.Hour * 4
+
+// TTLItem keeps association between serial number and ttl
+type TTLItem struct {
+	serialNumber storj.SerialNumber
+	ttl          time.Time
+}
+
+type createRequest struct {
+	Expiration *timestamp.Timestamp
+	Redundancy *pb.RedundancyScheme
+
+	ttl time.Time
+}
+
+type createRequests struct {
+	mu sync.RWMutex
+	// orders limit serial number used because with CreateSegment we don't have path yet
+	requests map[storj.SerialNumber]*createRequest
+
+	ttlMu    sync.Mutex
+	ttlItems []*TTLItem
+}
+
+func newCreateRequests() *createRequests {
+	return &createRequests{
+		requests: make(map[storj.SerialNumber]*createRequest),
+		ttlItems: make([]*TTLItem, 0),
+	}
+}
+
+func (cr *createRequests) Put(serialNumber storj.SerialNumber, createRequest *createRequest) {
```

Comment:
According our current coding style name suggestion, it looks that `cr` should be of a more explicit name.

Reply:
made some changes, hope is not worse :)